tests/attacks/test_knockoff_nets.py

```diff
@@ -0,0 +1,434 @@
+        model.add(Dense(10, input_shape=(4,), activation='relu'))
+        model.add(Dense(10, activation='relu'))
+        model.add(Dense(3, activation='softmax'))
+        model.compile(loss='categorical_crossentropy', optimizer=keras.optimizers.Adam(lr=0.001), metrics=['accuracy'])
+
+        # Get classifier
+        thieved_krc = KerasClassifier(model, clip_values=(0, 1), use_logits=False, channel_index=1)
+
+        # Create random attack
+        attack = KnockoffNets(classifier=victim_krc, batch_size_fit=BATCH_SIZE, batch_size_query=BATCH_SIZE,
+                              nb_epochs=NB_EPOCHS, nb_stolen=NB_STOLEN, sampling_strategy='random')
+        thieved_krc = attack.extract(x=self.x_train, thieved_classifier=thieved_krc)
+
+        victim_preds = np.argmax(victim_krc.predict(x=self.x_train[:100]), axis=1)
+        thieved_preds = np.argmax(thieved_krc.predict(x=self.x_train[:100]), axis=1)
+        acc = np.sum(victim_preds == thieved_preds) / len(victim_preds)
+
+        self.assertGreater(acc, 0.3)
+
+        # Create adaptive attack
+        attack = KnockoffNets(classifier=victim_krc, batch_size_fit=BATCH_SIZE, batch_size_query=BATCH_SIZE,
+                              nb_epochs=NB_EPOCHS, nb_stolen=NB_STOLEN, sampling_strategy='adaptive', reward='all')
+        thieved_krc = attack.extract(x=self.x_train, y=self.y_train, thieved_classifier=thieved_krc)
+
+        victim_preds = np.argmax(victim_krc.predict(x=self.x_train[:100]), axis=1)
+        thieved_preds = np.argmax(thieved_krc.predict(x=self.x_train[:100]), axis=1)
+        acc = np.sum(victim_preds == thieved_preds) / len(victim_preds)
+
+        self.assertGreater(acc, 0.4)
+
+        # Clean-up
+        k.clear_session()
+
+    def test_iris_pt(self):
+        """
+        Third test for Pytorch.
+        :return:
+        """
+        # Build PyTorchClassifier
+        victim_ptc = get_iris_classifier_pt()
+
+        class Model(nn.Module):
+            """
+            Create Iris model for PyTorch.
+            """
+
+            def __init__(self):
+                super(Model, self).__init__()
+
+                self.fully_connected1 = nn.Linear(4, 10)
+                self.fully_connected2 = nn.Linear(10, 10)
+                self.fully_connected3 = nn.Linear(10, 3)
+
+            # pylint: disable=W0221
+            # disable pylint because of API requirements for function
+            def forward(self, x):
+                x = self.fully_connected1(x)
+                x = self.fully_connected2(x)
+                logit_output = self.fully_connected3(x)
+
+                return logit_output
+
+        # Define the network
+        model = Model()
+
+        # Define a loss function and optimizer
+        loss_fn = nn.CrossEntropyLoss()
+        optimizer = optim.Adam(model.parameters(), lr=0.001)
+
+        # Get classifier
+        thieved_ptc = PyTorchClassifier(model=model, loss=loss_fn, optimizer=optimizer, input_shape=(4,),
```

Comment:
Would it be possible to create thieved_ptc with get_classifier_pt?

Reply:
Solved.